usl_pipeline/cloud_functions/main.py

```diff
@@ -307,12 +308,38 @@ def _read_wps_features(fd: IO[bytes]) -> Tuple[NDArray, FeatureMetadata]:
     # Ignore type checker error - BytesIO inherits from expected type BufferedIOBase
     # https://shorturl.at/lk4om
     with xarray.open_dataset(fd) as ds:  # type: ignore
-        feature = ds.data_vars["SNOALB"].values
-        # Drop time axis - each WPS output file will correspond to single time value
-        feature = numpy.squeeze(feature, axis=0)
+        features_matrix = numpy.array([])
+        for var in wps_variables.REQUIRED_VARS:
+            try:
+                feature = ds.data_vars[var].values
+                feature = _process_wps_feature(feature)
+
+                if not features_matrix.size:
+                    features_matrix = feature
+                else:
+                    features_matrix = numpy.dstack((features_matrix, feature))
+            except KeyError:
+                logging.warning(f"No variable named '{var}' found in dataset.")
```

Comment:
We'd need to stop the show here rather than log & continue. ML will only be happy if the tensor is of the exact expected dimensions, so we'd just be deferring an error until training or prediction time.

Reply:
Makes sense - remove the try/catch - will let the keyerror stop processing